pkg/services/ngalert/state/cache.go

```diff
@@ -92,7 +92,10 @@ func (c *cache) expandRuleLabelsAndAnnotations(alertRule *ngModels.AlertRule, la
 	expand := func(original map[string]string) map[string]string {
 		expanded := make(map[string]string, len(original))
 		for k, v := range original {
-			ev, err := expandTemplate(alertRule.Title, v, labels, values)
+			ev, err := c.expandTemplate(alertRule.Title, v, labels, values)
+			if ev == "" {
+				continue
```

Comment:
I think we do but I'm open to differing ideas. The reason I think we want to remove the key is that the key was not present in the response from the datasource. If we keep the template string, we're altering the datasource response. This can give us collisions in labelsets, too, which means that we don't really know what we're processing/returning in our API. I think that by removing keys that were not returned from the datasource, we are making the best effort to evaluate and alert on the data we have.